src/IO/Archives/LibArchiveReader.cpp

```diff
@@ -14,35 +12,57 @@ namespace DB
 
 namespace ErrorCodes
 {
-    extern const int CANNOT_UNPACK_ARCHIVE;
-    extern const int LOGICAL_ERROR;
-    extern const int CANNOT_READ_ALL_DATA;
-    extern const int UNSUPPORTED_METHOD;
+extern const int CANNOT_UNPACK_ARCHIVE;
+extern const int LOGICAL_ERROR;
+extern const int CANNOT_READ_ALL_DATA;
+extern const int UNSUPPORTED_METHOD;
 }
 
+class LibArchiveReader::StreamInfo
+{
+public:
+    explicit StreamInfo(std::unique_ptr<SeekableReadBuffer> read_buffer_) : read_buffer(std::move(read_buffer_)) { }
+
+    static ssize_t read([[maybe_unused]] struct archive * a, void * client_data, const void ** buff)
```

Comment:
let's not use names like `a` and change it to something more descriptive.

Reply:
Sorry I missed this, that is my bad. Fixing it.